fullstop-web/fullstop-timemachine/src/main/java/org/zalando/stups/fullstop/controller/WhitelistController.java

```diff
@@ -0,0 +1,73 @@
+package org.zalando.stups.fullstop.controller;
+
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.zalando.fullstop.web.api.NotFoundException;
+import org.zalando.stups.fullstop.rule.entity.RuleDTO;
+import org.zalando.stups.fullstop.rule.entity.RuleEntity;
+import org.zalando.stups.fullstop.rule.service.RuleEntityService;
+
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+import static org.springframework.web.bind.annotation.RequestMethod.*;
+
+@RestController
+@RequestMapping(value = "/whitelisting-rules", produces = APPLICATION_JSON_VALUE)
+@PreAuthorize("#oauth2.hasScope('uid')")
+public class WhitelistController {
+
+    @Autowired
+    RuleEntityService ruleEntityService;
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+
+    //TODO Swagger for all (?)
+    @RequestMapping(value = "/", method = GET)
+    @PreAuthorize("#oauth2.hasScope('uid')")
```

Comment:
I would let the `hasScope('uid')` as is. And additionally inject the `AuthenticationPrincipal` as method parameter and check back with the team API. A similar approach can be found at the violations controller, for resolving violations